devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/dependency/ConditionalDependenciesEnabler.java

```diff
@@ -102,12 +102,6 @@ private void collectConditionalDependencies(Set<ResolvedArtifact> runtimeArtifac
                         queueConditionalDependency(extension, conditionalDep);
                     }
                 }
-
-                // If the extension doesn't have any conditions we just enable it by default
-                if (extension.getDependencyConditions().isEmpty()) {
-                    extension.setConditional(true);
-                    enableConditionalDependency(extension.getExtensionId());
-                }
```

Comment:
After all, this seems to be the only change related to dependency processing?

Reply:
Yes, a simple fix for a misunderstanding on my part.